service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -510,22 +518,56 @@ public Language getImageLanguage() {
     public List<Representative> getRepresentativesByServedPreference(RepresentativeServingPreferences preference) {
         return getRepresentativesElementsByServedPreference(preference).stream()
             .map(Element::getValue)
-            .collect(Collectors.toList());
+            .filter(representative -> !getInactiveRepresentatives().contains(representative))
+            .collect(toList());
     }
 
     @JsonIgnore
     public List<Element<Representative>> getRepresentativesElementsByServedPreference(
-        RepresentativeServingPreferences preference
-    ) {
+        RepresentativeServingPreferences preference) {
         if (isNotEmpty(representatives)) {
             return representatives.stream()
                 .filter(Objects::nonNull)
+                .filter(representative -> !getInactiveRepresentatives().contains(representative.getValue()))
```

Comment:
might be nicer to store the result of `getInactiveRepresentatives()` before the stream so that we aren't constantly computing it